### ISSUE TITLE
Error: No version of module jupyter-matplotlib is registered

### ISSUE THREAD

**Issue by jpgomess** [microsoft/vscode-jupyter#16021]
## Environment data

-   VS Code version: ---
-   Jupyter Extension version (available under the Extensions sidebar): v2024.8.0
-   Python Extension version (available under the Extensions sidebar): v2024.14.0
-   OS (Windows | Mac | Linux distro) and version: Windows Server 2016
-   Python and/or Anaconda version: Python 3.9.19
-   Type of virtual environment used (N/A | venv | virtualenv | conda | ...): conda
-   Jupyter server running: Local

I am trying to plot an interactive plot in a jupyter notebook in a Conda Environment with the VSCode extension. The following error is given when trying to use the matplotlib backend **widget**:

`Error: No version of module jupyter-matplotlib is registered
    at ph.loadClass (https://file+.vscode-resource.vscode-cdn.net/c%3A/Users/jpg/.vscode/extensions/ms-toolsai.jupyter-renderers-1.0.19/out/node_modules/%40vscode/jupyter-ipywidgets8/dist/ipywidgets.js:2:4099813)
    at ph.loadClass (https://file+.vscode-resource.vscode-cdn.net/c%3A/Users/jpg/.vscode/extensions/ms-toolsai.jupyter-renderers-1.0.19/out/node_modules/%40vscode/jupyter-ipywidgets8/dist/ipywidgets.js:2:4403287)
    at ph.loadModelClass (https://file+.vscode-resource.vscode-cdn.net/c%3A/Users/jpg/.vscode/extensions/ms-toolsai.jupyter-renderers-1.0.19/out/node_modules/%40vscode/jupyter-ipywidgets8/dist/ipywidgets.js:2:4097773)
    at ph._make_model (https://file+.vscode-resource.vscode-cdn.net/c%3A/Users/jpg/.vscode/extensions/ms-toolsai.jupyter-renderers-1.0.19/out/node_modules/%40vscode/jupyter-ipywidgets8/dist/ipywidgets.js:2:4094616)
    at ph.new_model (https://file+.vscode-resource.vscode-cdn.net/c%3A/Users/jpg/.vscode/extensions/ms-toolsai.jupyter-renderers-1.0.19/out/node_modules/%40vscode/jupyter-ipywidgets8/dist/ipywidgets.js:2:4092246)
    at ph.handle_comm_open (https://file+.vscode-resource.vscode-cdn.net/c%3A/Users/jpg/.vscode/extensions/ms-toolsai.jupyter-renderers-1.0.19/out/node_modules/%40vscode/jupyter-ipywidgets8/dist/ipywidgets.js:2:4091039)
    at https://file+.vscode-resource.vscode-cdn.net/c%3A/Users/jpg/.vscode/extensions/ms-toolsai.jupyter-renderers-1.0.19/out/node_modules/%40vscode/jupyter-ipywidgets8/dist/ipywidgets.js:2:4402511
    at n._handleCommOpen (https://file+.vscode-resource.vscode-cdn.net/c%3A/Users/jpg/.vscode/extensions/ms-toolsai.jupyter-2024.8.0-win32-x64/dist/webviews/webview-side/ipywidgetsKernel/ipywidgetsKernel.js:3:80955)
    at async n._handleMessage (https://file+.vscode-resource.vscode-cdn.net/c%3A/Users/jpg/.vscode/extensions/ms-toolsai.jupyter-2024.8.0-win32-x64/dist/webviews/webview-side/ipywidgetsKernel/ipywidgetsKernel.js:3:82830)`

When %matplotlib inline is used instead no error occurs.

Already tried to create a new environment, to use a different computer, to paste and run the notebook file in another folder, but nothing happens...

The following packages are installed (Python=3.9.19):

`Package                      Version
---------------------------- --------------
absl-py                      2.1.0
addict                       2.4.0
anyio                        4.4.0
argon2-cffi                  23.1.0
argon2-cffi-bindings         21.2.0
arrow                        1.3.0
asciitree                    0.3.3
asttokens                    2.4.1
astunparse                   1.6.3
async-lru                    2.0.4
attrs                        23.2.0
babel                        2.16.0
balanced-loss                0.1.0
beautifulsoup4               4.12.3
bleach                       6.1.0
blinker                      1.7.0
cachetools                   5.3.3
certifi                      2024.2.2
cffi                         1.17.1
charset-normalizer           3.1.0
click                        8.0.4
cloudpickle                  2.2.1
colorama                     0.4.6
comm                         0.2.2
ConfigArgParse               1.7
contourpy                    1.2.1
cycler                       0.12.1
dash                         2.16.1
dash-core-components         2.0.0
dash-html-components         2.0.0
dash-table                   5.0.0
dask                         2024.4.1
dask-expr                    1.0.11
dask-glm                     0.3.2
dask-ml                      2024.4.4
debugpy                      1.8.1
decorator                    5.1.1
defusedxml                   0.7.1
distributed                  2024.4.1
exceptiongroup               1.2.0
executing                    2.0.1
fasteners                    0.19
fastjsonschema               2.19.1
filelock                     3.14.0
Flask                        3.0.3
flatbuffers                  24.3.25
fonttools                    4.51.0
fqdn                         1.5.1
fsspec                       2024.3.1
gast                         0.4.0
google-auth                  2.29.0
google-auth-oauthlib         0.4.6
google-pasta                 0.2.0
graphviz                     0.20.3
grpcio                       1.62.1
h11                          0.14.0
h5netcdf                     1.3.0
h5py                         3.11.0
httpcore                     1.0.5
httpx                        0.27.2
idna                         3.6
imageio                      2.34.0
importlib_metadata           7.1.0
importlib_resources          6.4.0
intel-openmp                 2021.4.0
ipykernel                    6.29.3
ipympl                       0.9.4
ipython                      8.18.1
ipython_genutils             0.2.0
ipywidgets                   8.1.2
isoduration                  20.11.0
itsdangerous                 2.1.2
jedi                         0.19.1
Jinja2                       3.1.3
joblib                       1.4.2
json5                        0.9.25
jsonpointer                  3.0.0
jsonschema                   4.21.1
jsonschema-specifications    2023.12.1
jupyter_client               8.6.1
jupyter_core                 5.7.2
jupyter-events               0.10.0
jupyter-lsp                  2.2.5
jupyter_server               2.14.2
jupyter_server_terminals     0.5.3
jupyterlab                   4.2.5
jupyterlab_pygments          0.3.0
jupyterlab_server            2.27.3
jupyterlab_widgets           3.0.10
keras                        2.10.0
Keras-Preprocessing          1.1.2
kiwisolver                   1.4.5
lazy_loader                  0.4
libclang                     18.1.1
llvmlite                     0.42.0
locket                       1.0.0
Markdown                     3.6
markdown-it-py               3.0.0
MarkupSafe                   2.1.5
matplotlib                   3.8.4
matplotlib-inline            0.1.6
mdurl                        0.1.2
mistune                      3.0.2
mkl                          2021.4.0
ml-dtypes                    0.4.0
more-itertools               10.2.0
mpmath                       1.3.0
msgpack                      1.0.8
multipledispatch             1.0.0
munkres                      1.1.4
namex                        0.0.8
nbclient                     0.10.0
nbconvert                    7.16.4
nbformat                     5.10.4
nest_asyncio                 1.6.0
networkx                     3.2.1
nodejs                       0.1.1
notebook_shim                0.2.4
numba                        0.59.1
numcodecs                    0.12.1
numpy                        1.23.5
oauthlib                     3.2.2
open3d                       0.18.0
opencv-python                4.9.0.80
opt-einsum                   3.3.0
optional-django              0.1.0
optree                       0.11.0
overrides                    7.7.0
packaging                    24.0
pandas                       2.2.1
pandocfilters                1.5.1
parso                        0.8.4
partd                        1.4.1
pickleshare                  0.7.5
pillow                       10.3.0
pip                          24.2
platformdirs                 4.2.0
plotly                       5.20.0
prometheus_client            0.20.0
prompt-toolkit               3.0.42
protobuf                     3.19.6
psutil                       5.9.8
pure-eval                    0.2.2
pyarrow                      16.0.0
pyasn1                       0.6.0
pyasn1_modules               0.4.0
pycparser                    2.22
pydot                        3.0.1
Pygments                     2.17.2
pyparsing                    3.1.2
python-dateutil              2.9.0
python-json-logger           2.0.7
pytz                         2024.1
pywin32                      306
pywinpty                     2.0.13
PyYAML                       6.0.1
pyzgy                        0.0.9
pyzmq                        25.1.2
referencing                  0.34.0
requests                     2.31.0
requests-oauthlib            2.0.0
retrying                     1.3.4
rfc3339-validator            0.1.4
rfc3986-validator            0.1.1
rich                         13.7.1
rpds-py                      0.18.0
rsa                          4.9
scikeras                     0.13.0
scikit-image                 0.22.0
scikit-learn                 1.4.2
scipy                        1.13.0
segyio                       1.9.12
segysak                      0.4
Send2Trash                   1.8.3
setuptools                   68.2.2
setuptools-scm               7.1.0
six                          1.16.0
sniffio                      1.3.1
sortedcontainers             2.4.0
soupsieve                    2.6
sparse                       0.15.1
stack-data                   0.6.2
sympy                        1.12
tbb                          2021.12.0
tblib                        3.0.0
tenacity                     8.2.3
tensorboard                  2.10.1
tensorboard-data-server      0.6.1
tensorboard-plugin-wit       1.8.1
tensorflow                   2.10.1
tensorflow-estimator         2.10.0
tensorflow-io-gcs-filesystem 0.31.0
termcolor                    2.4.0
terminado                    0.18.1
threadpoolctl                3.5.0
tifffile                     2024.2.12
tinycss2                     1.3.0
tomli                        2.0.1
toolz                        0.12.1
torch                        2.3.0
tornado                      6.4
tqdm                         4.66.2
traitlets                    5.14.2
types-python-dateutil        2.9.0.20240906
typing_extensions            4.11.0
tzdata                       2024.1
unicodedata2                 15.1.0
uri-template                 1.3.0
urllib3                      2.2.1
wcwidth                      0.2.13
webcolors                    24.8.0
webencodings                 0.5.1
websocket-client             1.8.0
Werkzeug                     3.0.2
wheel                        0.41.2
widgetsnbextension           4.0.10
wrapt                        1.16.0
xarray                       2024.3.0
zarr                         2.17.2
zict                         3.0.0
zipp                         3.17.0`

![image](https://github.com/user-attachments/assets/703d5842-4c92-4f31-9c6d-c2dd6e5b37e0)

![image](https://github.com/user-attachments/assets/eefdbca9-e7b3-4211-a152-72a2fd6bb857)

**Comment by DonJayamanne** [comment]
Duplicate of https://github.com/microsoft/vscode-jupyter/issues/16007

I'm sorry you are running into this issue, we've identified the issue and fixed it in the pre-release version.
Please install the latest pre-release version of the Jupyter extension to get around this issue
